src/edusharing-server/server.ts

```diff
@@ -0,0 +1,363 @@
+import express, { Request, Response } from 'express'
+import multer from 'multer'
+import * as t from 'io-ts'
+import { testContent } from './test-content'
+import { imageEmbedJson } from './mocked-embed-json/image'
+import { v4 as uuid_v4 } from 'uuid'
+import * as jose from 'jose'
+import urlJoin from 'url-join'
+import { readEnvVariable } from '../backend/read-env-variable'
+import { createAutoFromResponse } from '../backend/create-auto-form-response'
+
+export const editorUrl = readEnvVariable('EDITOR_URL')
+
+export const edusharingMockClientId = 'edusharing-mock'
+
+const VersionComment = t.union([t.null, t.string, t.array(t.string)])
+
+export class EdusharingServer {
+  private keys = jose.generateKeyPair('RS256', {
+    modulusLength: 2048,
+  })
+  private keyId = uuid_v4()
+  private state = '2452454263425'
+  private nonce = '8356345643564'
+  private defaultCustom = {
+    getContentApiUrl:
+      'http://localhost:8100/edu-sharing/rest/ltiplatform/v13/content',
+    fileName: 'Test Content',
+    getDetailsSnippetUrl:
+      'http://localhost:8100/edu-sharing/rest/lti/v13/details',
+    dataToken:
+      'kOXGc6AbqYW7iHOl3b48Pj/ngudoLCZk+DJwYxAg9wTiKsN9TKRY13qU+6vNNMEV2Guya3NPWO+Ay8IJDtQWMKxnkku/3mc+n64TIgMjs2yY7wXMYcvoRK4C9iXXpydNWQCGreYU2BcnMwne/b5BngOvBjqqVCPLMGT/lmvylP//GCzM7V99h9fKVMrgY97qOdsB1O0Ti//E3odWU1dFUMu3NLPy3MdTHXdViQpyPFRpgnZ8kcywDl0bLYSKy0pUuJy0hBvlnGmFyKlcQ38HaR2CZ9wRxrNgRxxEzGd8J+T6YSNoD8OyB9Nyjbp0N3tog4XhEZ/UASIqLYBzk+ygOA==',
+    postContentApiUrl:
+      'http://localhost:8100/edu-sharing/rest/ltiplatform/v13/content',
+    appId: 'qsa2DgKBJ2WgoJO1',
+    nodeId: '604f62c1-6463-4206-a571-8c57097a54ae',
+    user: 'admin',
+  }
+  private user = 'admin'
+  private custom = this.defaultCustom
+  private app = express()
+  private content: unknown = testContent
+  public savedVersions: Array<{ comment: t.TypeOf<typeof VersionComment> }> = []
+  private loginHint = uuid_v4()
+  private contextId = uuid_v4()
+
+  constructor() {
+    this.app.use(express.json())
+    this.app.use(express.urlencoded({ extended: true }))
+
+    this.app.get('/', (_req, res) => {
+      createAutoFromResponse({
+        res,
+        targetUrl: urlJoin(editorUrl, 'lti/login'),
```

Comment:
There probably is a package we could use. I have not looked yet, just ported the code over from `serlo-editor-for-edusharing`

Reply:
Not sure. But, I had to debug two times already to figure out why things did not work and in both cases a `/` was missing when building the URL. So, I figured I try to prevent this type of error. Especially since we will most likely change endpoints through `.env` again in the near future.